Add better description of what to pass to `hostfxr_initialize_for_dotnet_command_line` (#53678)
dotnet/runtime: docs/design/features/native-hosting.md

Initializes the hosting components for running a managed application.
The command line is parsed to determine the app path. The app path will be used to locate the `.runtimeconfig.json` and the `.deps.json` which will be used to load the application and its dependent frameworks.

> * `argc` and `argv` - the command line for running a managed application. These represent the arguments which would have been passed to the muxer if the app was being run from the command line. These are the parameters which are only valid for the runtime installation - SDK/CLI commands are not supported. So for example `app.dll app_argument_1 app_argument_2`. This specifically doesn't support the `dotnet run` SDK command.

Reviewer: ```suggestion
* `argc` and `argv` - the command line for running a managed application. These represent the arguments which would have been passed to the muxer if the app was being run from the command line. These are the parameters which are valid for the runtime installation by itself - SDK/CLI commands are not supported. For example, the arguments could be `app.dll app_argument_1 app_argument_2`. This API specifically doesn't support the `dotnet run` SDK command.
```

hostfxr_handle host_context_handle;
hostfxr_initialize_for_dotnet_command_line(
    _argc_,

>     _argv_,  // Should be for example 'app.dll app_argument_1 app_argument_2'

Reviewer: ```suggestion
    _argv_,  // For example, 'app.dll app_argument_1 app_argument_2'
```
